File: examples/python/best-practices/good-batch-operations.py
@durable_step
def process_batch(step_context: StepContext, items: list) -> list:
    return [process_item(item) for item in items]

@durable_execution
def lambda_handler(event: dict, context: DurableContext) -> list:
    items = event["items"]
    results = []
    
    # Process 10 items per step instead of 1
    for i in range(0, len(items), 10):
        batch = items[i:i+10]
        batch_results = context.step(
            process_batch(batch),
            name=f"process_batch_{i//10}"
        )
        results.extend(batch_results)
    
    return results

File: examples/java/best-practices/good-batch-operations.java
var items = (List<Map<String, Object>>) input.get("items");
var results = new ArrayList<>();

// Process 10 items per step instead of 1
for (int i = 0; i < items.size(); i += 10) {
    final var batch = items.subList(i, Math.min(i + 10, items.size()));
    var batchResults = ctx.step(
        "process_batch_" + (i / 10),
        new TypeToken<List<Object>>() {},
        stepCtx -> batch.stream().map(item -> processItem(item)).toList()
    );
    results.addAll(batchResults);
}
return results;
